File: DevEnvironment/Json/src/test/java/json/core/TestData.java
package json.core;

import base.core.ArrayLists;
import base.core.Pair;
import base.interfaces.IPair;
import json.interfaces.ITestData;
import testbase.core.ResourcePaths;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;

/**
 * The TestData class implements the data of the tests.
 */
public final class TestData implements ITestData {
    private static final String DATA_DIRECTORY_NAME = "data";
    private static final String JSON_DIRECTORY_NAME = "json";

    private static final List<IPair<String, Class<?>>> jsonResourcesInformation = ArrayLists.make(
        Pair.make("propertyType.json", PropertyType.class),
        Pair.make("fruit.json", Fruit.class),
        Pair.make("colors.json", Colors.class));

    /**
     * The TestData constructor.
     */
    public TestData() {
    }

    /**
     * Gets information of json resources.
     */
    @Override
    public List<IPair<Path, Class<?>>> getJsonResourcesInformation() {
        List<IPair<Path, Class<?>>> information = new ArrayList<>();

        for (IPair<String, Class<?>> resourceInformation : jsonResourcesInformation) {
            Path path = ResourcePaths.create(
                DATA_DIRECTORY_NAME,
                JSON_DIRECTORY_NAME,
                resourceInformation.first());

            Class<?> classType = resourceInformation.second();

            information.add(Pair.make(path, classType));
        }

        return information;
    }

}

File: DevEnvironmentCLion/JsonTest/core/TestData.cpp
#include "PreCompiled.h"

#include "TestData.h"
#include "ClassType.h"
#include "ResourcePaths.h"
#include "ClassType.h"

using namespace test::json;

const std::string TestData::dataDirectoryName = "data";
const std::string TestData::jsonDirectoryName = "json";
std::vector<std::pair<std::string, base::ClassType>> TestData::jsonResourcesInformation =
{
    std::make_pair(
        "colors.json",
        ::base::ClassType("Colors")),
    std::make_pair(
        "fruit.json",
        ::base::ClassType("Fruit")),
    std::make_pair(
        "propertyType.json",
        ::base::ClassType("PropertyType"))
};

/**
 * The TestData constructor.
 */
TestData::TestData()
{
}

/**
 * The TestData destructor.
 */
TestData::~TestData()
{
}

/**
 * Gets information of json resources.
 */
void TestData::getJsonResourcesInformation(std::vector<std::pair<std::string, ::base::ClassType>>& result)
{
    for (const std::pair<std::string, ::base::ClassType>& resourceInformation : jsonResourcesInformation)
    {
        for (const std::pair<std::string, ::base::ClassType>& resourceInformation : jsonResourcesInformation)
        {
            std::string path = test::base::ResourcePaths::create(
                dataDirectoryName,
                jsonDirectoryName,
                resourceInformation.first);

            const ::base::ClassType& classType = resourceInformation.second;
            result.push_back(std::make_pair(path, classType));
        }
    }
}
